rsc/src/main/java/com/cloudera/livy/rsc/rpc/Rpc.java

```diff
@@ -275,10 +273,14 @@ public void operationComplete(ChannelFuture cf) {
       };
 
       dispatcher.registerRpc(id, promise, msg.getClass().getName());
-      synchronized (channelLock) {
```

Comment:
From my understanding, we already have a lock to avoid race condition, do we need to put into `eventloop`?